examples/graphics/hardware-instancing.html

```diff
@@ -87,38 +87,40 @@
         // add the box entity to the hierarchy
         app.root.addChild(box);
 
-        // number of instances to render
-        var instanceCount = 1000;
-
-        // store matrices for individual instances into array
-        var matrices = new Float32Array(instanceCount * 16);
-        var matrixIndex = 0;
-
-        var radius = 5;
-        var pos = new pc.Vec3();
-        var rot = new pc.Quat();
-        var scl = new pc.Vec3();
-        var matrix = new pc.Mat4();
-
-        for (var i = 0; i < instanceCount; i++) {
-            // generate random positions / scales and rotations
-            pos.set(Math.random() * radius - radius * 0.5, Math.random() * radius - radius * 0.5, Math.random() * radius - radius * 0.5);
-            scl.set(0.1 + Math.random() * 0.1, 0.1 + Math.random() * 0.3, 0.1 + Math.random() * 0.1);
-            rot.setFromEulerAngles(i * 30, i * 50, i * 70);
-            matrix.setTRS(pos, rot, scl);
-
-            // copy matrix elements into array of floats
-            for (var m = 0; m < 16; m++)
-                matrices[matrixIndex++] = matrix.data[m];
+        if (this.app.graphicsDevice.supportsInstancing) {
```

Comment:
what does the example do now if instancing isn't supported?

Reply:
renders the background/mini-stats. The same as transform-feedback.